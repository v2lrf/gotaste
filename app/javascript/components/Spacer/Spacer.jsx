import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

const sizes = [
  '0',
  '1',
  '2',
  '3',
  '4',
  '5',
  '6',
  '8',
  '10',
  '12',
  '16',
  '20',
  '24',
  '32',
  false
]

function Spacer({
  children,
  top,
  right,
  bottom,
  left,
  vertical,
  horizontal,
  inner,
  display
}) {
  const spacerType = inner ? 'p' : 'm'
  const spacerClasses = classnames({
    [`${spacerType}t-${top}`]: top,
    [`${spacerType}r-${right}`]: right,
    [`${spacerType}b-${bottom}`]: bottom,
    [`${spacerType}l-${left}`]: left,
    [`${spacerType}y-${vertical}`]: vertical,
    [`${spacerType}x-${horizontal}`]: horizontal
  })

  return display === 'inline' ? (
    <span className={spacerClasses}>{children}</span>
  ) : (
    <div className={spacerClasses}>{children}</div>
  )
}

Spacer.propTypes = {
  children: PropTypes.node.isRequired,
  top: PropTypes.oneOf(sizes),
  right: PropTypes.oneOf(sizes),
  bottom: PropTypes.oneOf(sizes),
  left: PropTypes.oneOf(sizes),
  vertical: PropTypes.oneOf(sizes),
  horizontal: PropTypes.oneOf(sizes),
  inner: PropTypes.bool,
  display: PropTypes.oneOf(['block', 'inline'])
}

Spacer.defaultProps = {
  display: 'block',
  top: false,
  right: false,
  bottom: false,
  left: false,
  vertical: false,
  horizontal: false,
  inner: false
}

export default Spacer
