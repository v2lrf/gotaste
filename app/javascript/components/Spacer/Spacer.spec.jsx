import React from 'react'
import Spacer from './Spacer'

describe('Spacer', () => {
  let component
  beforeEach(() => {
    component = shallow(<Spacer>I am inside a spacer!</Spacer>)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
