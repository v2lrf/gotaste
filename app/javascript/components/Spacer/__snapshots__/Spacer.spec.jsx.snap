// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer Snapshots should match the snapshot 1`] = `
<div
  class=""
>
  I am inside a spacer!
</div>
`;
